cli/pcluster/config/config_patch.py

```diff
@@ -0,0 +1,256 @@
+                    self._compare_section(base_section, target_section)
+
+    def _compare_section(self, base_section, target_section):
+        """
+        Compare the provided base and target sections and append the detected changes to the internal changes list.
+
+        :param base_section: The section in the base configuration
+        :param target_section: The corresponding section in the target configuration
+        """
+        for _, param in target_section.params.items():
+            base_value = base_section.get_param_value(param.key)
+            target_value = param.value
+
+            if not self._compare_params(base_section.key, param.key, base_value, target_value):
+                self.changes.append(
+                    Change(
+                        target_section.key,
+                        target_section.label,
+                        param.key,
+                        base_value,
+                        param.value,
+                        param.get_update_policy(),
+                    )
+                )
+
+    def _compare_params(self, section_key, param_key, base_value, target_value):
+        """
+        Compare the specified base configuration parameter with the corresponding one in target configuration.
+
+        The way the comparison is performed depends on the availability of a custom comparator for the specified
+        parameter. If no custom comparator is available, a default equality comparison is performed.
+
+        :param section_key: The section key
+        :param param_key: The parameter key
+        :param base_value: The value of the parameter in the base configuration
+        :param target_value: The value of the parameter in the target configuration
+        :return: True if the values can be considered equal, False otherwise
+        """
+        comparator = ConfigPatch.PARAM_COMPARATORS.get((section_key.lower(), param_key.lower()))
+        if comparator:
+            result = comparator(base_value, target_value)
+        else:
+            result = base_value == target_value
+        return result
+
+    def _remove_ignored_sections(self, config):
+        for section_key in ConfigPatch.IGNORED_SECTIONS:
+            config.remove_section(section_key)
+
+    @property
+    def update_policy_level(self):
+        """
+        Get the max update policy level of the ConfigPatch.
+
+        This method provides a static indication of the actual applicability of the patch and it's meant for testing
+        purposes. The real applicability of a patch, which involves dynamic condition checks, must be performed
+        by calling the check() method.
+        """
+        return (
+            max(change.update_policy.level for change in self.changes)
+            if len(self.changes)
+            else UpdatePolicy.SUPPORTED.level
+        )
+
+    def _create_default_section(self, config, section):
```

Comment:
It's a static method, the `self` object is not used here, I'd move it outside the `ConfigPatch` object.
In future it could became an utility method to create default sections and we could move it in the `param_types.py` module for example.

Reply:
The logic of this method is very specific to ConfigPatch class. It creates a section with all default values with the same id and label of the provided one. It is meant to create a mirror section to compare with the existing one. We could make it static and also rename it but I'd leave it inside ``ConfigPatch.``